types/module/server/msgs.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// isServiceMsg checks if a type URL corresponds to a service method name,
+// i.e. /cosmos.bank.Msg/Send vs /cosmos.bank.MsgSend.
+// Function copied from cosmos-sdk, once we add ADR 033 and group module to it,
+// we'll want to merge them.
+func isServiceMsg(typeURL string) bool {
+	return strings.Count(typeURL, "/") >= 2
+}
+
+// SetMsgs takes a slice of sdk.Msg's and turn them into Any's.
+// This is similar to what is in the cosmos-sdk tx builder
+// and could eventually be merged in.
+func SetMsgs(msgs []sdk.Msg) ([]*types.Any, error) {
+	anys := make([]*types.Any, len(msgs))
+	for i, msg := range msgs {
+		var err error
+		switch msg := msg.(type) {
+		case sdk.ServiceMsg:
+			anys[i], err = types.NewAnyWithCustomTypeURL(msg.Request, msg.MethodName)
+		default:
+			anys[i], err = types.NewAnyWithValue(msg)
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+	return anys, nil
+}
+
+// SetMsgs takes a slice of Any's and turn them into sdk.Msg's.
+// This is similar to what is in the cosmos-sdk sdk.Tx
+// and could eventually be merged in.
+func GetMsgs(anys []*types.Any) []sdk.Msg {
+	msgs := make([]sdk.Msg, len(anys))
+	for i, any := range anys {
+		var msg sdk.Msg
+		if isServiceMsg(any.TypeUrl) {
+			req := any.GetCachedValue()
+			if req == nil {
+				panic("Any cached value is nil. Transaction messages must be correctly packed Any values.")
```

Comment:
yeah, it's better to return error than repeating the pattern from SDK.

Nodes won't crash, because there is a high level `recover` (try-catch) to handle all that panics. There are tonnes of panics in SDK codebase, because the early developers preferred to panic and catch (recover) rather than chaining errors.

Reply:
sure I can update to return an error instead, although these functions are ultimately used in `Get...` functions that will panic in case of error.